tests/core/test_core.py

```diff
@@ -187,7 +187,6 @@ def test_bash_operator_kill(self):
             self.fail("BashOperator's subprocess still running after stopping on timeout!")
 
     def test_on_failure_callback(self):
-        # Annoying workaround for nonlocal not existing in python 2
```

Comment:
Can we also remove the workaround itself?

Reply:
I don't think it's really workaround here. The current test case requires the `on_failure_callback` to change the value of `data['called']` and it doesn't require nonlocal to do that.

Reply:
@uranusjr what do you think is the workaround here?

Reply:
I this it is a workaround in the sense that this is not straightforward at first glance without a comment (why does the variable needs to be a dict, looks like `called = False` would work). So if we're removing the comment, it's probably better to use `nonlocal` since it makes it clearer why the straightforward implementation does not work.

Reply:
Ahh, it makes sense. After some thought, I think using `MagicMock` for the callback and then call `assert_called_once` would be a better alternative.

Reply:
Better indeed 👍

Reply:
This is nice